spec.emu

```diff
@@ -1108,18 +1108,37 @@ contributors: Ron Buckton, Ecma International
       </h1>
       <dl class="header"></dl>
       <emu-alg>
-        1. For each _resource_ of _disposeCapability_.[[DisposableResourceStack]], in reverse list order, do
-          1. Let _result_ be Dispose(_resource_.[[ResourceValue]], _resource_.[[Hint]], _resource_.[[DisposeMethod]]).
-          1. If _result_.[[Type]] is ~throw~, then
-            1. If _completion_.[[Type]] is ~throw~, then
-              1. Set _result_ to _result_.[[Value]].
-              1. Let _suppressed_ be _completion_.[[Value]].
-              1. Let _error_ be a newly created *SuppressedError* object.
-              1. Perform CreateNonEnumerableDataPropertyOrThrow(_error_, *"error"*, _result_).
-              1. Perform CreateNonEnumerableDataPropertyOrThrow(_error_, *"suppressed"*, _suppressed_).
-              1. Set _completion_ to ThrowCompletion(_error_).
+        1. Let _needsAwait_ be *false*.
+        1. Let _hasAwaited_ be *false*.
+        1. For each element _resource_ of _disposeCapability_.[[DisposableResourceStack]], in reverse list order, do
+          1. Let _value_ be _resource_.[[ResourceValue]].
+          1. Let _hint_ be _resource_.[[Hint]].
+          1. Let _method_ be _resource_.[[DisposeMethod]].
+          1. If _hint_ is ~sync-dispose~ and _needsAwait_ is *true* and _hasAwaited_ is *false*, then
+            1. Perform ! Await(*undefined*).
+            1. Set _needsAwait_ to *false*.
+            1. Set _hasAwaited_ to *false*.
```

Comment:
Is this line necessary? `_hasAwaited_` should already be `*false*` according to the `If`